firebase.go

```diff
@@ -64,17 +64,18 @@ func New(url string) *Firebase {
 	tr = &http.Transport{
 		DisableKeepAlives: true, // https://code.google.com/p/go/issues/detail?id=3514
 		Dial: func(network, address string) (net.Conn, error) {
-			start := time.Now()
-			c, err := net.DialTimeout(network, address, TimeoutDuration)
-			tr.ResponseHeaderTimeout = TimeoutDuration - time.Since(start)
-			return c, err
+			return net.DialTimeout(network, address, TimeoutDuration)
 		},
+		ResponseHeaderTimeout: TimeoutDuration,
 	}
 
 	return &Firebase{
 		url:          sanitizeURL(url),
 		params:       _url.Values{},
-		client:       &http.Client{Transport: tr},
+		client:       &http.Client{
+			Transport: tr,
+			Timeout:   TimeoutDuration,
```

Comment:
Looking at the docs for [http.Client](https://golang.org/pkg/net/http/#Client) we could probably just get away with using the `Timeout` field and ditch the transport all together.